recipes/minisat/all/conanfile.py

```diff
@@ -0,0 +1,85 @@
+from conan import ConanFile
+from conan.tools.apple import fix_apple_shared_install_name
+from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain, cmake_layout
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.files import apply_conandata_patches, collect_libs, copy, export_conandata_patches, get
+from conan.tools.microsoft import is_msvc
+from os.path import join
+
+required_conan_version = ">=1.53.0"
+
+
+class MiniSatConan(ConanFile):
+    name = "minisat"
+    description = "minimalistic SAT solver"
+    license = "MIT"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "http://minisat.se"
+    topics = ("satisfiability", "solver")
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
+
+    def validate(self):
+        if is_msvc(self) and self.options.shared:
+            raise ConanInvalidConfiguration(f"{self.ref} can not be built as shared on Visual Studio and msvc.")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def export_sources(self):
+        export_conandata_patches(self)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def requirements(self):
+        self.requires("zlib/1.2.13", transitive_headers=True)
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["STATIC_BINARIES"] = not self.options.shared
+        tc.variables["USE_SORELEASE"] = False
+        tc.generate()
+        tc = CMakeDeps(self)
+        tc.generate()
+
+    def _determine_lib_name(self):
+        return f"minisat-lib-{'shared' if self.options.shared else 'static'}"
+
+    def build(self):
+        apply_conandata_patches(self)
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build(target=self._determine_lib_name())
+
+    def package(self):
```

Comment:
Is its not possible to use a cmake install?

Reply:
CMake install will install the static and the shared lib. Thus, the recipe would need to compile both and would install both. That also is true for the command line binary.

Reply:
```suggestion
    def package(self):
         # Manually copy since install does both static and shared 
         # https://github.com/conan-io/conan-center-index/pull/16071#discussion_r1149458542
```